apps/www/_blog/2023-12-15-introducing-read-replicas.mdx

```diff
@@ -0,0 +1,141 @@
+
+### File-Based Log shipping 🤝 Streaming replication
+
+We use a hybrid approach to address the limitations of each method.
+
+<Img
+  src={{
+    dark: '/images/blog/lwx-read-replicas/streaming-replication-dark.png',
+    light: '/images/blog/lwx-read-replicas/streaming-replication-light.png',
+  }}
+  alt="Streaming replication"
+/>
+
+Streaming replication minimizes replication lag, while file-based log shipping provides a fallback. For file-based log shipping, we use our existing Point In Time Recovery (PITR) infrastructure. We regularly archive files from the primary using [WAL-G](https://github.com/wal-g/wal-g), an open source archival and restoration tool, and ship the WAL files to S3.
+
+We combine it with streaming replication to reduce replication lag. Once WAL-G files have been synced from S3, read replicas connect to the primary and stream the WAL directly.
+
+## What's coming
+
+### Automatic Failover
+
+With automatic failovers, a read replica is promoted to be the primary if the primary is unhealthy. This feature is currently only available on our enterprise plan. However, we are planning to extend this to all paid plans. The introduction of self-serve read replicas is a step in that direction.
+
+### Geo-based load balancing
+
+Currently, we have implemented a round-robin strategy of routing PostgREST requests to the different databases. This is useful for sharing the load across the primary and the read replica, but in cases where latency is important, you might want to always route to the closest database instead.
+
+### Supabase Products leveraging Replicas
+
+We are working on support for database-level load balancing through [Supavisor](https://supabase.com/blog/supabase-branching). When routing database traffic through Supavisor, it routes write queries to the primary and splits read queries between the primary and read replica. This distributes load to all your databases without any changes to your application. Other Supabase products like Auth, Storage, and Realtime use the database heavily. We are working on adding read replica support to these products, so that they can leverage the read replica when appropriate. This means they won’t just query the primary but will also use the read replica’s compute when possible.
```

Comment:
Here we are linking to branching post, should it be to the supavisor post?

Reply:
updating

Reply:
updated